### ISSUE TITLE
[feature] Would like a status bar for uploading conan packages 

### ISSUE THREAD

**Issue by timviola-cmi** [conan-io/conan#15122]
### What is your suggestion?

# Request Description

It would be really helpful if there were a real time updating status bar indicating the progress and speed/rate of conan package uploads for users. This lets the user know if their system is frozen, or if the server is responding, and approximately how long the upload will take, or if there are issues with the connection speed.

# Prototype work

For my own purposes I inserted a simple prototype of the desired functionality into conans/client/rest/file_uploader.py into the function `_upload_file` using the tqdm library. Introducing this kind of dependency into conan is probably undesirable, but the result is a good example of what is being requested.

```python
def _upload_file(self, url, abs_path,  headers, auth):
        with open(abs_path, mode='rb') as file_handler:
            try:
                filesize = os.path.getsize(abs_path)
                response = self._requester.put(url, 
                                               data=tqdm(file_handler, 
                                                         total=filesize, 
                                                         unit='B',unit_scale=True), 
                                               verify=self._verify_ssl,
                                               headers=headers, auth=auth)
                self._handle_400_response(response, auth)
                response.raise_for_status()  # Raise HTTPError for bad http response status
                return response
            except ConanException:
                raise
            except Exception as exc:
                raise ConanException(exc) 
```

## Prototype Behavior

![image](https://github.com/conan-io/conan/assets/115578984/c08ca239-8fdd-44bb-af4c-a76db78db448)

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by tbsuht** [comment]
Related in terms of showing progress: https://github.com/conan-io/conan/issues/14589

**Comment by memsharded** [comment]
Hi @timviola-cmi 

Thanks for your suggestion.
There are 2 different aspects here. The first on, graphical progress bars, we already had them in Conan 1.X. They were quite problematic, so we decided to drop them in Conan 2. They were simply not worth it, making things more fragile and complicated, specially for parallel things. There were a bunch of bugs and issues in Conan 1 because the progress bars made uploads fail, for example.

A different story is that we want to show progress somehow. So we started to introduce messages reporting progress. At the moment, upload of big packages should start reporting their % of progress every 10 seconds or so. We thought that was enough at the moment.

The next improvement will likely be https://github.com/conan-io/conan/issues/14589, as @tbsuht commented, because that is lacking.

From there, we can further improve the progress information, parameterizing the intervals for messages or adding more information like transfer speed, etc. I think this is a valid request and can be done. But not in the form of progress bars, I don't think we will introduce them back in Conan 2.

**Comment by timviola-cmi** [comment]
Hello @memsharded. Thanks for following up and explaining the difficulties/tradeoffs involved. 

Uploads of big packages reporting their progress every 10 seconds would be great, that does not appear to be working for me or anyone else in my organization right now. I'll keep looking into it on my end to see what is happening there, I wasn't aware it was supposed to do that. We are uploading large packages to a test server running an Artifactory CE instance at the moment and it sits for many minutes without any indication of any kind. 

You can close this issue if you'd like or leave it open, but if we keep it open, we can treat it as an issue to add transfer speed to progress messages (or I can open a new issue).

**Comment by memsharded** [comment]
> Uploads of big packages reporting their progress every 10 seconds would be great, that does not appear to be working for me or anyone else in my organization right now. 

What Conan version are you using? This was added in some of the recent ones.

**Comment by timviola-cmi** [comment]
@memsharded This issue is coming up again (we had paused our usage of Artifactory, but have now resumed) so I'm posting here with some more detail. I am using conan 2.2.2 and uploading to an Artifactory Community Edition 7.63.12 rev 76312900 server. The package I'm uploading is 1.1GB in size. The conan upload command was static for several minutes with no progress indication. Then after about 10minutes displayed the message "upload completed in 624s". It seems like this is not the intended behavior, what is the best way to proceed?

![image](https://github.com/user-attachments/assets/3cf494c6-b71d-48f7-a2dc-2fe115fa53a6)


![image](https://github.com/user-attachments/assets/c797acc0-400e-491a-8ac2-3302acb8a057)

**Comment by memsharded** [comment]
Sorry there was some confusion above, the progress was added for other parts (downloads).
PR https://github.com/conan-io/conan/pull/16913 is adding support for uploads.

In any case, I think that 624s for uploading 1.1Gb sounds a bit too slow, like 1Mb/s, it seems that it should be possible to upload faster than that, not sure where the bottleneck could be, but it might be worth investigating.
I'll ask for some benchmarks for uploads in ConanCenter tomorrow.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/16913 was merged, it will be released next Conan 2.8
